3rd-party/woocommerce.php

```diff
@@ -1,14 +1,137 @@
 <?php
 
-add_action( 'woocommerce_share', 'jetpack_woocommerce_social_share_icons', 10 );
 
-/*
- * Make sure the social sharing icons show up under the product's short description
- */
-function jetpack_woocommerce_social_share_icons() {
-	if ( function_exists( 'sharing_display' ) ) {
-		remove_filter( 'the_content', 'sharing_display', 19 );
-		remove_filter( 'the_excerpt', 'sharing_display', 19 );
-		echo sharing_display();
+class Jetpack_WooCommerce {
+	public static function init() {
+		add_action( 'woocommerce_share', array( 'Jetpack_WooCommerce', 'social_share_icons' ), 10 );
+		add_filter( 'jetpack_options_whitelist', array( 'Jetpack_WooCommerce', 'options_whitelist' ), 10 );
+		add_filter( 'jetpack_sync_constants_whitelist', array( 'Jetpack_WooCommerce', 'constants_whitelist' ), 10 );
+		add_filter( 'jetpack_sync_post_meta_whitelist', array( 'Jetpack_WooCommerce', 'meta_meta_whitelist' ), 10 );
 	}
+
+	/*
+	 * Make sure the social sharing icons show up under the product's short description
+	 */
+	public static function social_share_icons() {
+		if ( function_exists( 'sharing_display' ) ) {
+			remove_filter( 'the_content', 'sharing_display', 19 );
+			remove_filter( 'the_excerpt', 'sharing_display', 19 );
+			echo sharing_display();
+		}
+	}
+
+	public static function options_whitelist( $list ) {
+		self::add_to_whitelist( $list, self::$wc_options_whitelist );
+	}
+
+	public static function constants_whitelist( $list ) {
+		self::add_to_whitelist( $list, self::$wc_constants_whitelist );
+	}
+
+	public static function meta_meta_whitelist( $list ) {
+		self::add_to_whitelist( $list, self::$wc_post_meta_whitelist );
+	}
+
+	private static function add_to_whitelist( $whitelist, $values ) {
```

Comment:
Wouldn't this be simpler to just use `array_merge()` instead?

https://secure.php.net/manual/en/function.array-merge.php